IntellitectTerminal.Web/src/views/Home.vue

```diff
@@ -4,121 +4,139 @@
-            cursorPosition++;
-        }
-      })
+      this.initTerminal(input)
+    }
+  }
+
+  initTerminal(input: HTMLElement) {
+    this.term.open(input);
+
+    // MOTD
+    this.term.write(this.welcomeMessage);
+    this.term.write("\r\n");
+
+    // Log path you are on
+    this.term.write(this.path);
+
+    // Main key handler. Anything pressed goes here.
+    this.term.onKey((e) => this.keyPressedHandler(e));
+  }
+
+  keyPressedHandler(event: { key: any; domEvent?: KeyboardEvent; }) {
+
+    // Command keys
+    switch (event.key) {
+
+      case Keys.BACKSPACE:
+        // If the cursor is going to delete from our path... dont
+        if (this.cursorPosition == this.path.length) { return; }
+        this.term.write("\b \b");
+        this.userInput = this.userInput.substring(0, this.userInput.length - 1);
+        this.cursorPosition--;
+        return;
+
+      case Keys.ENTER:
+        this.term.write(event.key);
+        this.userInput += event.key;
+        this.term.write("\n");
+
+        this.commandHandler(this.userInput.trim());
+
+        this.term.write(this.path);
+        this.userInput = "";
+        this.cursorPosition = this.path.length;
+        return;
+
+      case Keys.ARROW_LEFT:
+        // If the cursor is going out of our text... dont
+        if (this.cursorPosition == this.path.length) { return; }
+        this.term.write("\x1B[D");
+        this.cursorPosition--;
+        return;
+
+      case Keys.ARROW_RIGHT:
+        // If the cursor is going out of our text... dont
+        if (this.cursorPosition >= (this.path.length + this.userInput.length)) { return; }
+        this.term.write("\x1B[C");
+        this.cursorPosition++;
+        return;
+
+      case Keys.ARROW_UP:
+        break;
+
+      case Keys.ARROW_BOTTOM:
+        break;
+
+      default:
+        console.log(event);
+        this.term.write(event.key);
+        this.userInput += event.key;
+        this.cursorPosition++;
+    }
+  }
 
+  commandHandler(cmd: string) {
+    switch (cmd) {
+      case Commands.HELP:
+        this.term.write(" help - Displays this message");
+        this.term.write("\r\n");
+        this.term.write(" ls - Lists all files in the current directory");
+        this.term.write("\r\n");
+        this.term.write(" cat - Displays the contents of a file");
+        this.term.write("\r\n");
+        this.term.write(" mkdir - Creates a directory");
+        this.term.write("\r\n");
+        this.term.write(" challenge - Requests a challenge");
+        this.term.write("\r\n");
+        this.term.write(" submit - Submits a challenge");
+        this.term.write("\r\n");
+        this.term.write(" edit - Edits a file");
+        this.term.write("\r\n");
+        this.term.write(" progress - Displays your progress");
+        this.term.write("\r\n");
+        this.term.write(" verify - Verifies a challenge");
+        this.term.write("\r\n");
+        this.term.write('\x1B[1;3;31mIntelliTect\x1B[0m $ ');
```

Comment:
```suggestion
        this.term.write(this.path);
```
Use this.path instead of explicit string